src/main/java/com/github/rakhmedovrs/avro/utils/AvroUtils.java

```diff
@@ -0,0 +1,34 @@
+package com.github.rakhmedovrs.avro.utils;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.*;
+import org.apache.avro.specific.SpecificData;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+/**
+ * @author Ruslan Rakhmedov
+ * @created 1/21/2024
+ */
+public class AvroUtils {
+	public static <T extends GenericRecord> T parseJson(String jsonString, Schema schema)
```

Comment:
I think there is important part missing - Class<T> that is required to have typed datum reader.
It could look like this:

`	public static <T extends GenericRecord> T parseJson(String jsonString, Class<T> clazz, Schema schema) throws IOException {
		SpecificDatumReader<T> reader = new SpecificDatumReader<>(clazz);
		JsonDecoder jsonDecoder = DecoderFactory.get().jsonDecoder(schema, jsonString);
		return reader.read(null, jsonDecoder);
	}`

Reply:
Fixed